src/main/java/com/taskadapter/redmineapi/internal/RequestParam.java

```diff
@@ -0,0 +1,43 @@
+package com.taskadapter.redmineapi.internal;
+
+import java.util.Objects;
+
+public class RequestParam {
+
+    private final String name;
+    private final String value;
+
+    public RequestParam(final String name, final String value) {
+        this.name = Objects.requireNonNull(name, "Name may not be null");
+        this.value = Objects.requireNonNull(value, "Value may not be null");
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        RequestParam that = (RequestParam) o;
+        return name.equals(that.name);
+    }
```

Comment:
this may be confusing to people. why are params with the same name considered "equal"? what about the value?

if some part of the code wants to treat these Params as "equal" as long as they have equal names, it is better to keep this comparison logic close to that area, maybe with a local Comparator class or something, but not inside the data object itself. this would allow reusability for this class in the future.